pyomo/contrib/preprocessing/plugins/var_aggregator.py

```diff
@@ -248,6 +255,12 @@ def _apply_to(self, model, detect_fixed_vars=True):
             # the variables in its equality set.
             z_agg.setlb(max_if_not_None(v.lb for v in eq_set if v.has_lb()))
             z_agg.setub(min_if_not_None(v.ub for v in eq_set if v.has_ub()))
+            # Set the domain of the aggregate variable to the intersection of
+            # the domains of the variables in its equality set
+            domain = Reals
+            for v in eq_set:
+                domain = domain & v.domain
```

Comment:
This isn't particularly efficient, but it is clean and correct.

Reply:
OK, this makes sense. I think I'll merge as-is because if we ever want more efficiency, we will want to do a rewrite anyway, I think.